packages/connector-azuread/src/index.ts

```diff
@@ -26,54 +25,34 @@ import {
   AzureADConfig,
   accessTokenResponseGuard,
   userInfoResponseGuard,
+  authResponseGuard,
 } from './types';
 
-export default class AzureADConnector implements SocialConnectorInstance<AzureADConfig> {
-  public metadata: ConnectorMetadata = defaultMetadata;
+// eslint-disable-next-line @silverhand/fp/no-let
+let clientApplication: ConfidentialClientApplication;
```

Comment:
This won't work as expected.

`clientApplication` can not be reused, because the `config` may change, we should create `clientApplication` each time, and then we no longer need to disable this lint rule.

Reply:
Can do it later

Reply:
Sounds solid. As all other connectors' requests which rely on DB config fetch up-to-date DB config in advance, we should also align this to Azure AD connector.
Add [LOG-4012](https://linear.app/silverhand/issue/LOG-4012/azure-ad-connector-should-not-reuse-confidentialclient) to track this.